src/sage/schemes/weighted_projective/weighted_projective_point.py

```diff
@@ -0,0 +1,297 @@
+r"""
+Points on projective schemes
+
+This module implements scheme morphism for points on weighted projective schemes.
+
+AUTHORS:
+
+- Gareth Ma (2024): initial version, based on unweighted version.
+"""
+
+# ****************************************************************************
+#       Copyright (C) 2006 David Kohel <kohel@maths.usyd.edu.au>
+#       Copyright (C) 2006 William Stein <wstein@gmail.com>
+#       Copyright (C) 2011 Volker Braun <vbraun.name@gmail.com>
+#       Copyright (C) 2024 Gareth Ma <grhkm21@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#                  https://www.gnu.org/licenses/
+# ****************************************************************************
+
+from sage.categories.fields import Fields
+from sage.rings.fraction_field import FractionField
+from sage.misc.misc_c import prod
+
+from sage.schemes.generic.morphism import (SchemeMorphism,
+                                           SchemeMorphism_point)
+from sage.structure.sequence import Sequence
+from sage.structure.richcmp import richcmp, op_EQ, op_NE
+
+
+# --------------------
+# Projective varieties
+# --------------------
+
+class SchemeMorphism_point_weighted_projective_ring(SchemeMorphism_point):
+    """
+    A rational point of weighted projective space over a ring.
+
+    INPUT:
+
+    - ``X`` -- a homset of a subscheme of an ambient weighted projective space over a ring `R`.
+
+    - ``v`` -- a list or tuple of coordinates in `R`.
+
+    - ``check`` -- boolean (default: ``True``). Whether to check the input for consistency.
+
+    EXAMPLES::
+
+        sage: WP = WeightedProjectiveSpace(2, ZZ)
+        sage: WP(2, 3, 4)
+        (2 : 3 : 4)
+    """
+
+    def __init__(self, X, v, check=True):
```

Comment:
```suggestion
    def __init__(self, X, v, check: bool = True):
```
Let's use type annotations, at least for easy cases like basic Python types.